Tests/test_ttry.py

```diff
@@ -0,0 +1,141 @@
+        node = {
+            "description": "ttry",
+            "elements": {
+                "e1": {
+                    "description": "traise",
+                    "elements": {
+                        "e1": "raise"
+                    }
+                },
+                "e2": {
+                    "description": "tvar",
+                    "elements": {
+                        "e1": "x"
+                    }
+                }
+            }
+        }
+
+        assert t_try({"x": TYPE.BOOL}, node) == TYPE.BOOL
+
+    # try true with raise
+    def test_t_try1(self):
+        node = {
+            "description": "ttry",
+            "elements": {
+                "e1": {
+                    "description": "tbool",
+                    "elements": {
+                        "e1": "true"
+                    }
+                },
+                "e2": {
+                    "description": "traise",
+                    "elements": {
+                        "e1": "raise"
+                    }
+                }
+            }
+        }
+
+        assert t_try({"x": TYPE.BOOL}, node) == TYPE.BOOL
+
+    # try raise with raise
+    def test_t_try2(self):
+        node = {
+            "description": "ttry",
+            "elements": {
+                "e1": {
+                    "description": "traise",
+                    "elements": {
+                        "e1": "raise"
+                    }
+                },
+                "e2": {
+                    "description": "traise",
+                    "elements": {
+                        "e1": "raise"
+                    }
+                }
+            }
+        }
+
+        assert t_try({}, node) == TYPE.UNDEFINED
+
+    # try x with true
+    def test_t_try3(self):
+        node = {
+            "description": "ttry",
+            "elements": {
+                "e1": {
+                    "description": "tbool",
+                    "elements": {
+                        "e1": "true"
+                    }
+                },
+                "e2": {
+                    "description": "tvar",
+                    "elements": {
+                        "e1": "x"
+                    }
+                }
+            }
+        }
+
+        assert t_try({"x": TYPE.BOOL}, node) == TYPE.BOOL
+
+    # try x with true
+    # x not valid
+    def test_t_try4(self):
+        node = {
+            "description": "ttry",
+            "elements": {
+                "e1": {
+                    "description": "tbool",
+                    "elements": {
+                        "e1": "true"
+                    }
+                },
+                "e2": {
+                    "description": "tvar",
+                    "elements": {
+                        "e1": "x"
+                    }
+                }
+            }
+        }
+
+        assert t_try({}, node) == TYPE.ERROR
+
+    # try true with blaa
+    # blaa not valid
+    def test_t_try5(self):
```

Comment:
Poderia testar também alguma com função
Do tipo ( ((int->int)->bool) raise) ver se ela consome o parâmetro 
(ja que supostamente é o comportamento esperado)

Reply:
Andy, eu não to fazendo o app, eu fiz o try raise!